apps/gitness/src/pages/pull-request/stores/pull-request-store.tsx

```diff
@@ -0,0 +1,255 @@
+import { create } from 'zustand'
+import { produce } from 'immer'
+import { mergePullReqOp, commentStatusPullReq as apiCommentStatusPullReq } from '@harnessio/code-service-client'
+import { CodeCommentState, PullRequestState } from '../types/types'
+import { PullRequestDataState } from '@harnessio/views'
+export const codeOwnersNotFoundMessage = 'CODEOWNERS file not found'
+export const codeOwnersNotFoundMessage2 = `path "CODEOWNERS" not found`
+export const codeOwnersNotFoundMessage3 = `failed to find node 'CODEOWNERS' in 'main': failed to get tree node: failed to ls file: path "CODEOWNERS" not found`
+export const oldCommitRefetchRequired = 'A newer commit is available. Only the latest commit can be merged.'
+export const prMergedRefetchRequired = 'Pull request already merged'
+export const POLLING_INTERVAL = 10000
+
+export const usePullRequestDataStore = create<PullRequestDataState>((set, get) => ({
+  repoMetadata: undefined,
+  setRepoMetadata: metadata =>
+    set(
+      produce(draft => {
+        draft.repoMetadata = metadata
+      })
+    ),
+  pullReqMetadata: undefined,
+  pullReqStats: undefined,
+  pullReqCommits: undefined,
+  pullReqActivities: undefined,
+  loading: false,
+  error: null,
+  pullReqChecksDecision: {
+    overallStatus: undefined,
+    count: {
+      error: 0,
+      failure: 0,
+      pending: 0,
+      running: 0,
+      success: 0,
+      skipped: 0,
+      killed: 0
+    },
+    error: null,
+    data: undefined,
+    color: '',
+    background: '',
+    message: '',
+    summaryText: '',
+    checkInfo: {
+      title: '',
+      content: '',
+      color: '',
+      status: ''
+    }
+  },
+  showEditDescription: false,
+  setShowEditDescription: show =>
+    set(
+      produce(draft => {
+        draft.showEditDescription = show
+      })
+    ),
+  setRuleViolationArr: arr =>
+    set(
+      produce(draft => {
+        draft.prPanelData.ruleViolationArr = arr
+      })
+    ),
+  refetchActivities: () => {},
+  refetchCommits: () => {},
+  refetchPullReq: () => {},
+  retryOnErrorFunc: () => {},
+  dryMerge: () => {
+    const { repoMetadata, pullReqMetadata, refetchPullReq } = get()
+    const isClosed = pullReqMetadata?.state === PullRequestState.CLOSED
+    if (!isClosed && repoMetadata?.path !== undefined && pullReqMetadata?.state !== PullRequestState.MERGED) {
+      mergePullReqOp({
+        repo_ref: `${repoMetadata?.path}/+`,
+        pullreq_number: Number(pullReqMetadata?.number),
+        body: { bypass_rules: true, dry_run: true, source_sha: pullReqMetadata?.source_sha }
+      })
+        .then(({ body: res }) => {
+          set(
+            produce(draft => {
+              if (res?.rule_violations?.length && res?.rule_violations?.length > 0) {
```

Comment:
Not a big deal, but you can simplify
```suggestion
              if (res?.rule_violations?.length) {
```

Reply:
**done**